### ISSUE TITLE
Allow infix syntax for interface names

### ISSUE THREAD

**Issue by mdimjasevic** [idris-lang/Idris-dev#4267]
Idris doesn't support infix syntax for interface names. In particular, I would expect the following to compile:

# Steps to Reproduce

```idris
infixr 7 :<:
interface (Functor sub, Functor sup) => (sub : Type -> Type) :<: (sup : Type -> Type) where
  inj : sub a -> sup a
```

# Expected Behavior

The above snippet should compile so it can be used like tihs:

```idris
val : (Val :<: f) => Int -> Expr f
```

# Observed Behavior

The snippet above doesn't compile. Idris 1.1.1-git:afa8a25d reports:

```
- + Errors (1)
 `-- ./DataTypes.idr line 54 col 40:
      error: expected: function name
     <terface (Functor sub, Functor sup) => (sub : Type -> Type) :<: (sup : Type ->>
                                            ^                                       
```

Equivalent Haskell code that compiles goes like this:

```haskell
class (Functor sub, Functor sup) => sub :≺: sup where
  inj :: sub a -> sup a
```

**Comment by Melvar** [comment]
Does it work if you use it prefix in the interface declaration? My guess is that infix doesn’t work because this part of the interface declaration is not actually an expression of any kind, but its own separate syntax.

**Comment by mdimjasevic** [comment]
Yes, this compiles when in the prefix notation. I didn't get what you're saying. From my understanding, it can work if defined and used in a certain way.

**Comment by melted** [comment]
This is expected behavior. Idris doesn't support infix interface names.

**Comment by ahmadsalim** [comment]
@melted I will retag at is a feature request 😄 .
